src/shared/tasks/sentryconfig/tasksentryconfig.cpp

```diff
@@ -0,0 +1,85 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "tasksentryconfig.h"
+
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QJsonValue>
+#include <QStringList>
+
+#include "appconstants.h"
+#include "leakdetector.h"
+#include "logger.h"
+#include "networkrequest.h"
+#include "sentry/sentryadapter.h"
+#include "settingsholder.h"
+
+namespace {
+Logger logger("TaskSentryConfig");
+}
+
+TaskSentryConfig::TaskSentryConfig() : Task("TaskSentryConfig") {
+  MZ_COUNT_CTOR(TaskSentryConfig);
+}
+
+TaskSentryConfig::~TaskSentryConfig() { MZ_COUNT_DTOR(TaskSentryConfig); }
+
+void TaskSentryConfig::run() {
+  NetworkRequest* request = new NetworkRequest(this, 200);
+
+  QUrl url(AppConstants::apiBaseUrl());
+  url.setPath("/api/v1/vpn/crashreporting");
+
+  connect(request, &NetworkRequest::requestFailed, this,
+          [this](QNetworkReply::NetworkError error, const QByteArray&) {
+            Q_UNUSED(error);
+            logger.error() << "Failed to get Sentry info";
+            emit completed();
+          });
+  connect(request, &NetworkRequest::requestCompleted, this,
+          [this](const QByteArray& data) {
+            if (handleNetworkResponse(data)) {
+              // We now should have sentry info - let's init
+              SentryAdapter::instance()->init();
+            }
+            emit completed();
+          });
+
+  request->get(url);
+}
+
+bool TaskSentryConfig::handleNetworkResponse(const QByteArray& data) {
+  auto doc = QJsonDocument::fromJson(data);
+  if (!doc.isObject()) {
+    logger.error() << "Failed to get Sentry info";
+    return false;
+  }
+  // The only thing right now, we don't want to send
+  // crash-data, so we will only check for that.
+  QJsonObject obj = doc.object();
+  if (!obj.contains("dsn")) {
+    logger.error() << "DSN missing from Sentry info";
+    return false;
+  };
+  auto dsn = obj["dsn"].toString();
+  if (dsn.isNull()) {
+    logger.error() << "DSN missing from Sentry info";
+    return false;
+  };
+
+  if (!obj.contains("endpoint")) {
+    logger.error() << "endpoint missing from Sentry info";
+    return false;
+  };
+  auto endpoint = obj["endpoint"].toString();
+  if (endpoint.isNull()) {
+    logger.error() << "endpoint missing from Sentry info";
+    return false;
+  };
+  auto settings = SettingsHolder::instance();
+  settings->setSentryEndpoint(endpoint);
+  settings->setSentryDSN(dsn);
```

Comment:
How come this is not constant

Reply:
Now currently you need to know our sentry creds to enable crash reporting. 
This is not good for linux, as when we build using PPA's we cannot have secrets :) 
But we also don't want that piece of info in our repo. 
That's why we moved that part to guardian and pull it from there.

Reply:
Ah ok, so the whole reason for this is Linux essentially? I get not wanting it on our repo, but it could have been a build var otherwise.

Reply:
> Ah ok, so the whole reason for this is Linux essentially? 
Yes. :D

Reply:
Ok, would you mind adding a comment explaining this on the code? I guess the definition of TaskSentryConfig is the best place for it.